### PR TITLE
Add the checkbox feature

### PR DESCRIPTION
**Pull request by khanvong-ork**
Add the X red icon when readonly and value is false and use the  default checkmark when value is true
image

### CODE REVIEW

**Comment by Anstsiya** [comment]
Seems you did no changes for unit tests. Update them as well please.

File: src/components/MaterialUI/Inputs/CheckboxProps.js [hunk 1]
```diff
@@ -5,13 +5,17 @@ class CheckboxProps extends ComponentProps {
 		update: "update",
 		value: "value",
 		label: "label",
+		readOnly: "readOnly",
+		checked: "checked",
```
Comment (joclefrancois) [inline src/components/MaterialUI/Inputs/CheckboxProps.js:9]: Why did you add "checked" as "value" is already there ?

Comment (Anstsiya) [reply src/components/MaterialUI/Inputs/CheckboxProps.js:9]: Seems that checked and value it's different props. Value is for the value checkbox represents, but checked for Boolean value of that value

Comment (joclefrancois) [reply src/components/MaterialUI/Inputs/CheckboxProps.js:9]: @Anstsiya, yes, but for the Wrapper, value was already fulfilling that purposes, we don't need that duplication.

Comment (Anstsiya) [reply src/components/MaterialUI/Inputs/CheckboxProps.js:9]: @joclefrancois  but don't we need to know what value was changed and put it into a handler to have it in edit mode?

Comment (joclefrancois) [reply src/components/MaterialUI/Inputs/CheckboxProps.js:9]: Furthermore, it is a breaking changes for any application using it already.

File: src/components/MaterialUI/Inputs/Checkbox.js [hunk 2]
```diff
@@ -44,35 +45,98 @@ const useStyles = makeStyles(theme => ({
...
 const Checkbox = ({ checkboxProps }) => {
 	if (isCheckboxProps(checkboxProps) === false) {
 		throw new TypeError("checkboxProps property is not of type CheckboxProps");
 	}
 
+	const checked = checkboxProps?.get(CheckboxProps.propNames.checked) || false;
 	const update = checkboxProps?.get(CheckboxProps.propNames.update);
 	const value = checkboxProps?.get(CheckboxProps.propNames.value);
 	const label = checkboxProps?.get(CheckboxProps.propNames.label);
+	const readOnly = checkboxProps?.get(CheckboxProps.propNames.readOnly) || false;
```
Comment (joclefrancois) [inline src/components/MaterialUI/Inputs/Checkbox.js:111]: You need to change the file "CheckboxProps" as well and its unit tests.

**Review by joclefrancois — CHANGES_REQUESTED** [review]
See me other comments

File: src/components/MaterialUI/Inputs/CheckboxProps.js (hunk 1 above)
Comment (khanvong-ork) [reply src/components/MaterialUI/Inputs/CheckboxProps.js]: Removed checked so we don't add possible breaking changes. We should have further discussion on why we need the checked attribute.

File: src/components/MaterialUI/Inputs/Checkbox.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import React from "react";
-import clsx from 'clsx';
+import clsx from "clsx";
```
Comment (joclefrancois) [inline src/components/MaterialUI/Inputs/Checkbox.js:2]: You don't use it

**Review by joclefrancois — APPROVED** [review]

File: src/components/MaterialUI/Inputs/Checkbox.js (hunk 3 above)
Comment (khanvong-ork) [reply src/components/MaterialUI/Inputs/Checkbox.js]: Removed the referenced

**Review by Crispried — APPROVED** [review]
